test/es7/PromiseRejectionTracking.js

```diff
@@ -0,0 +1,110 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+// Test HostPromiseRejectionTracker - see ecma262 section 25.4.1.9
+
+let tests = [
```

Comment:
A couple of cases that I don't see captured here, and that I have some questions about:

```js
let controller;
let promise = new Promise((resolve, reject) => {  controller = {resolve, reject};});
let a = promise.then(() => {});
let b = promise.then(() => {});
controller.reject("Should see 2 unhandled rejections");

let c = a.then(() => {}); // Not clear: Should this have one "handled" and one "unhandled rejection" fire?

c.catch(() => {}); // Should have at least one handled fire, maybe 2 if the previous line did not fire a handled event?
```

Reply:
@MSLaguana a.then() will fire one “handled” (for a) and one “rejected” (for c).  c.then() should fire another “handled” for c.

Reply:
That makes sense. I think it'd be good to add these cases to the test file just to be sure as well.

Reply:
I agree more cases may be a good idea - it gets quite messy working with the order of results due to the async nature of promises - the set I did was the most I could do without async results coming in in odd orders.

Note the above case gives just 1 notification:
As follows:
1. .reject() is called -> promise is handled (as the .then()s have occurred) - handling of .then()s scheduled for the next tick of the event loop (via promise continuation callback)
2. On reaching the event loop: a is handled by c. b -> rejected trigger-ring notification. c -> is handled by the .catch()

I can add some more cases before Monday (don't have much time today)

Reply:
@rhuanjl is right, the example is completely synchronous and I forgot the .then()s got deferred till next tick.  Async code is confusing sometimes 🤪

Reply:
Ah I see; and the tests that currently test the `handled` case only apply when the initial rejection immediately has no `then` and so the rejection notice is synchronous. I do think it'd be great if you could add a more complex test case in line with what I had above, but as you say I'm not sure how best to make it deterministic.

Reply:
Wouldn’t be too bad if I did one file per case for complex cases.

I’m without a computer until Sunday now though so can’t do this today. Could make it a follow up PR.

Reply:
I've now added some other more involved cases including the above.